dodiscover/constraint/skeleton.py

```diff
@@ -496,3 +513,119 @@ def _postprocess_ci_test(
             self.sep_set_[y_var][x_var].append(set(cond_set))
             return True
         return False
+
+
+class LearnSemiMarkovianSkeleton(LearnSkeleton):
+    """Learning a semi-markovian skeleton.
+
+    This proceeds by learning a skeleton by testing edges with candidate
+    separating sets from the "possibly d-separating" sets (PDS), or PDS
+    sets that lie on a path between two nodes. This algorithm requires
+    the input of a collider-oriented PAG, which provides the necessary
+    information to compute the PDS set for any given nodes.
+
+    Parameters
+    ----------
+    ci_estimator : BaseConditionalIndependenceTest
+        The conditional independence test function.
+    sep_set : dictionary of dictionary of list of set
+        Mapping node to other nodes to separating sets of variables.
+        If ``None``, then an empty dictionary of dictionary of list of sets
+        will be initialized.
+    alpha : float, optional
+        The significance level for the conditional independence test, by default 0.05.
+    min_cond_set_size : int
+        The minimum size of the conditioning set, by default 0. The number of variables
+        used in the conditioning set.
+    max_cond_set_size : int, optional
+        Maximum size of the conditioning set, by default None. Used to limit
+        the computation spent on the algorithm.
+    max_combinations : int, optional
+        Maximum number of tries with a conditioning set chosen from the set of possible
```

Comment:
Can you rephrase the definition of `max_combinations`.  I think I get it, but the language is awkward

Reply:
Done. Lmk wdyt?